.github/dependabot.yml

```diff
@@ -0,0 +1,15 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "docker" # See documentation for possible values
+    directory: "/containers/release-candidate/" # Location of package manifests
```

Comment:
What will this actually check in the release candidate container? Does it build/run it, or just look in the Dockerfile for and container versions that can be updated? For example, right now we use Fedora 34, will it just add a PR when Fedora 35 containers become available?

Reply:
Dependabot will just look in the Dockerfile for whatever it can update, e.g., it will submit a PR when a Fedora 35 container supersedes the Fedora 34 container.

Reply:
If this just changes the numbers, but this is only run when we do a release, then I would suggest we just let these PRs pile up, and accept them before a release, when they will then get tested as part of creating an RC1.

Reply:
I'd actually rather we get to this kind of change well before we are ready to do an RC to avoid any issues.

For example, when we updated to Fedora 34 recently there were a couple of issues that took a bit of work to track down and fix (e.g. dependency changes, missing container privileges). If we wait until we're ready for an RC, there's a good chance new issues will crop up and unnecessarily delay the release. Getting to it earlier gives us more time to get those issues fixed.

Reply:
I'm fine with that but accepting a PR of this kind of change implies that the person integrating it has tested that it in fact works. With the regular scalabot automated tests, our CI automatically will catch things that it breaks. So if the licenses are all ok still, once can pretty much accept them without fear. For this, nothing is going to automatically check anything. So you have to be willing to spend the time on creating a mini-release which creates the build container etc. 

That is unless we want to do automated CI for that too.

Reply:
Agreed, these changes definitely shouldn't be accepted without manual testing.

It might be interesting to add the release candidate script to CI. GitHub Actions does have a way to build Docker images to use in actions, though from a quick skim it seems like you need to publish the docker image, and then it can be used for an action. Need to look more into if it's possible to build a docker image and use it in the same CI run--we don't really want to publish this image.

Some poential issues with CI:
* Currently the script tries to publish everything which we don't want to do for a PR. But we can just use the --dry-run option to disable that, so not a big deal.
* The script expects user input. Either need a way to pass everything in as options, or maybe can we just write to stdin
* The script builds from apache/master. It can be configured to build from a different repo/branch, so we'd probably need to configure it to build from the PR branch. I think that information is available from github actions so shouldn't be hard.
* GPG signing requires keys--I guess we could generate keys for testing
* The script clones repositories with git+ssh, which requires keys that github knows about. Need a way to switch to http

Probably all doable, but requires a bit of work. For something that doesn't update very often, it might be best to just stick to manual testing when it's needed.

Reply:
Mike, I can add a `reviewers: [ mbeckerle, stevedlawrence, ... ]` to the docker package manager configuration, which should warn a person not to accept a PR for the release dockerfile unless it has passed manual testing by one of the people who actually have released Daffodil in the past.  Will that change get a +1 from you?

Reply:
I will approve this. It's effectively still an experiment. Ultimately my concerns are downstream someday, about how this affects our release workflow, etc.